File: app/src/main/java/com/handstandsam/shoppingapp/models/CategoryBuilder.java
package com.handstandsam.shoppingapp.models;


public class CategoryBuilder {

    Category category;

    public CategoryBuilder() {
        category = new Category(null, null, null);
    }

    public CategoryBuilder(String label) {
        this();
        label(label);
    }

    public CategoryBuilder label(String label) {
        category.setLabel(label);
        return this;
    }

    public Category build() {
        return category;
    }

    public CategoryBuilder image(String image) {
        category.setImage(image);
        return this;
    }

    public CategoryBuilder link(String link) {
        category.setLink(link);
        return this;
    }
}

File: app/src/main/java/com/handstandsam/shoppingapp/models/CategoryBuilder.kt
package com.handstandsam.shoppingapp.models


class CategoryBuilder() {

    internal var category: Category = Category(null, null, null)

    constructor(label: String) : this() {
        label(label)
    }

    fun label(label: String): CategoryBuilder {
        category.label = label
        return this
    }

    fun image(image: String): CategoryBuilder {
        category.image = image
        return this
    }

    fun link(link: String): CategoryBuilder {
        category.link = link
        return this
    }

    fun build(): Category {
        return category
    }

}
